earth_enterprise/src/fusion/khgdal/khGDALReader.cpp

```diff
@@ -91,6 +93,9 @@ khGDALReader::khGDALReader(const khGDALDataset &srcDS_, uint numbands)
   }
 }
 
+void khGDALReader::GetNoDataFromSrc(double & no_data, int & nodata_exists) {
+  no_data = srcDS->GetRasterBand(1)->GetNoDataValue(&nodata_exists);
```

Comment:
This assumes nodata is the same for all bands. It's actually moved from line 47 of khGDALReaderImpl.h (the old version), so the logic hasn't changed.

Reply:
In theory according to raster data and GDAL it could be different. That is why it's per band and not at dataset level. The likelihood that this happens with RGB images is small but it does happens with multi-spectral bands, terrain stored in multiple bands (16 bits, 32 bits float, etc.).

Reply:
Yes, the way Open GEE handles masking is very much a heuristic and there are lots of failure scenarios. Truly fixing it would be a significant undertaking because raster import and masking happen in different processes, so getting them to work together may require changing the format of data on disk, providing an upgrade path, etc. For this fix, the idea is that I'm not changing the way anything works except for one specific case that was obviously wrong. I've written #1648 to properly handle nodata and alpha bands. I think handling per-band nodata would fit into that issue nicely or a separate follow up.

Reply:
Okay. Sounds reasonable to go with the fix as is while we post-pond a more global one into the other ticket.